packages/ragbits-document-search/src/ragbits/document_search/documents/sources/web.py

```diff
@@ -9,84 +9,85 @@
     import aiohttp
 
 from ragbits.core.utils.decorators import requires_dependencies
-from ragbits.document_search.documents.exceptions import SourceNotFoundError
+from ragbits.document_search.documents.exceptions import SourceNotFoundError, WebDownloadError
 from ragbits.document_search.documents.sources import Source
 from ragbits.document_search.documents.sources.base import get_local_storage_dir
 
 
-class HttpSource(Source):
+class WebSource(Source):
     """
-    An object representing an HTTP dataset source.
+    An object representing a Web dataset source.
     """
 
-    url: str
+    uri: str
```

Comment:
Not that this is particularly important, but I wonder why? To me "URL" seems more appropriate in this case (i.e., needing a _location_ to fetch data from). "URI", on the other hand seem to broad since it may refer to URIs that do not specify locations - which we can't use here.